---
title: "cg-onesample"
author: "XSun"
date: "2020-09-15"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In this part, I split the DGN data into two halves, and use one half to train the expression model, and the other half to assess association with factors.

## Methods

1.  Weights training

Split the DGN samples(922) into two equal part (part1-training part, part2).  Employ FUSION to train weights on part1, models: LASSO, Elastic-net, BLUP

2. Cis-gene imputation 

Use weights in step1 to impute the cis genetic component part of gene expression. For both part1 and part2, we imputed cis-genes and named these cis-genes as cis-gene-part1 and cis-gene-part2.

3. Association test: 

LVs ~ cis-gene-part1

LVs ~ cis-gene-part2

Record the p-values for each gene.

4.Count the number of significant associations at various p-value threshold for each LV.

## Results

### Overview

1. 12,398 of 12,491 genes have cis-snps.

2. I got 5629 weights from lasso model,  5635 weights from blup models.

### Summary 

The tables below show the number of significant associations at various p-value threshold for each LV. The first column is the pearson corralation index. The suffix '_s' means impute gene expression using the same part samples and the suffix '_d' means impute gene expression using different part samples. 

I also made a histogram to show the distribution of the numbers.

<figure class="half">
    <img src="https://github.com/xsun1229/factor_analysis_DGN/raw/master/output/hist_ciscomp_pliercanon.jpeg" width="200%">
</figure>

- Lasso model

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis_DGN/output/compare_sum_lasso_pliercanon.rdata")
DT::datatable(summary, options = list(pageLength =10))
```

- Blup model 

```{r echo=FALSE}
load("/project2/xinhe/xsun/website/factor_analysis_DGN/output/compare_sum_blup_pliercanon.rdata")
DT::datatable(summary, options = list(pageLength =10))
```
